src/headers.rs

```diff
@@ -7,14 +7,14 @@ use reqwest::StatusCode;
 use std::collections::BTreeMap;
 use std::io::{BufWriter, Write};
 
-pub struct Headers<'a, 'b> {
+pub struct Headers<'a, 'b, W: Write> {
```

Comment:
Interestingly I didn't _HAVE_ to assign a trait bound here. 

I'm presuming the compiler was able to figure that out from the below `impl` block which _does_ use a trait bound (at the compiler's request).

That said, I decided to be explicit about the trait bound here.